packages/rainbowkit/src/components/ChainModal/ChainModal.tsx

```diff
@@ -1,33 +1,49 @@
-import React, { Fragment, useContext } from 'react';
-import { useDisconnect, useNetwork, useSwitchNetwork } from 'wagmi';
+import React, { useContext, useState } from 'react';
+import { useAccount, useDisconnect, useSwitchChain } from 'wagmi';
+import { useConfig } from 'wagmi';
 import { isMobile } from '../../utils/isMobile';
-import { AsyncImage } from '../AsyncImage/AsyncImage';
 import { Box } from '../Box/Box';
 import { CloseButton } from '../CloseButton/CloseButton';
 import { Dialog } from '../Dialog/Dialog';
 import { DialogContent } from '../Dialog/DialogContent';
 import { DisconnectSqIcon } from '../Icons/DisconnectSq';
 import { MenuButton } from '../MenuButton/MenuButton';
-import { AppContext } from '../RainbowKitProvider/AppContext';
 import { I18nContext } from '../RainbowKitProvider/I18nContext';
 import { useRainbowKitChains } from '../RainbowKitProvider/RainbowKitChainContext';
 import { Text } from '../Text/Text';
+import Chain from './Chain';
 import {
   DesktopScrollClassName,
   MobileScrollClassName,
 } from './ChainModal.css';
 
 export interface ChainModalProps {
   open: boolean;
+  // We will use this for only testing purposes since for mock accounts
+  // wagmi won't give use the "chainId"
+  mockChainId?: number;
   onClose: () => void;
 }
 
-export function ChainModal({ onClose, open }: ChainModalProps) {
-  const { chain: activeChain } = useNetwork();
-  const { chains, pendingChainId, reset, switchNetwork } = useSwitchNetwork({
-    onSettled: () => {
-      reset(); // reset mutation variables (eg. pendingChainId, error)
-      onClose();
+export function ChainModal({ onClose, open, mockChainId }: ChainModalProps) {
+  const { chainId: wagmiChainId } = useAccount();
+  const chainId = mockChainId || wagmiChainId;
+  const { chains } = useConfig();
+  const [pendingChainId, setPendingChainId] = useState<number | null>(null);
+  const { switchChain } = useSwitchChain({
+    mutation: {
+      onMutate: ({ chainId }) => {
+        setPendingChainId(chainId);
+      },
+      onSuccess: () => {
+        if (pendingChainId) setPendingChainId(null);
+      },
+      onError: () => {
+        if (pendingChainId) setPendingChainId(null);
+      },
+      onSettled: () => {
+        onClose();
+      },
```

Comment:
There is no lunger pendingChainId here. Since everything is migrated over to tanstack react query we would have to find out ourselves which chain id is pending. This seems to be working just fine.